web/app/static/js/controllers/clusterController.js

```diff
@@ -554,13 +554,13 @@ angular.module('uluwatuControllers').controller('clusterController', ['$scope',
                     $rootScope.activeClusterBlueprint = $filter('filter')($rootScope.blueprints, {
                         id: $rootScope.activeCluster.blueprintId.toString()
                     }, true)[0];
-                    if ($rootScope.activeCluster.credentialId) {
+                    if ((typeof($rootScope.activeCluster.cloudPlatform) === "undefined" || !$rootScope.activeCluster.cloudPlatform) && $rootScope.activeCluster.credentialId) {
                         $rootScope.activeClusterCredential = $filter('filter')($rootScope.credentials, {
                             id: $rootScope.activeCluster.credentialId
                         }, true)[0];
-                        $rootScope.activeCluster.cloudPlatform = $rootScope.activeClusterCredential.cloudPlatform
-                    } else {
-                        $rootScope.activeCluster.cloudPlatform = $rootScope.activeCluster.orchestrator.type
```

Comment:
This was included here because of Mesos - in that case cloudPlatform is generally "BYOS" but we don't want to show that but Marathon/Mesos instead.
The original condition may not be correct but the orchestrator.type should stay here.